src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -1269,23 +1348,26 @@ public static boolean hasItemHandlerToolWithLevel(@NotNull final IItemHandler it
      * @param sourceHandler The {@link IItemHandler} that works as Source.
      * @param sourceIndex   The index of the slot that is being extracted from.
      * @param targetHandler The {@link IItemHandler} that works as Target.
-     *
      * @return True when the swap was successful, false when not.
      */
-    public static boolean transferItemStackIntoNextFreeSlotInItemHandlers(@NotNull final IItemHandler sourceHandler, @NotNull int sourceIndex, @NotNull IItemHandler targetHandler) {
+    public static boolean transferItemStackIntoNextFreeSlotInItemHandlers(@NotNull final IItemHandler sourceHandler, @NotNull int sourceIndex, @NotNull IItemHandler targetHandler)
+    {
         ItemStack sourceStack = sourceHandler.extractItem(sourceIndex, Integer.MAX_VALUE, true);
-        final ItemStack originalStack = sourceStack.copy();
 
-        for (int i = 0; i < targetHandler.getSlots(); i++) {
+        for (int i = 0; i < targetHandler.getSlots(); i++)
+        {
             sourceStack = targetHandler.insertItem(i, sourceStack, false);
-            if (isItemStackEmpty(sourceStack)) {
+            if (isItemStackEmpty(sourceStack))
+            {
                 sourceHandler.extractItem(sourceIndex, Integer.MAX_VALUE, false);
                 return true;
             }
         }
 
-        if (!ItemStack.areItemStacksEqual(sourceStack, originalStack) && compareItemStacksIgnoreStackSize(sourceStack, originalStack)) {
-            int usedAmount = originalStack.stackSize - originalStack.stackSize;
+        final ItemStack originalStack = sourceStack.copy();
+        if (!ItemStack.areItemStacksEqual(sourceStack, originalStack) && compareItemStacksIgnoreStackSize(sourceStack, originalStack))
```

Comment:
Yep line 1367 needs to be above the iteration through the slots. The variable is there to check for a succesfull insertion of the stack.